### PR TITLE
fanctl: init at 0.6.4

### PR DESCRIPTION
**Pull request by icewind1991**
###### Description of changes

Add [fanctl](https://gitlab.com/mcoffin/fanctl), a fancontrol replacement.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/fanctl/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    owner = "mcoffin";
+    repo = pname;
+    rev = "fbb440c520630dff57a8d9923fc0c70ea549caa5";
+    sha256 = "sha256-XmawybmqRJ9Lj6ii8TZBFwqdQZVp0pOLN4xiSLkU/bw=";
+  };
+
+  cargoSha256 = "sha256-tj00DXQEqC/8+3uzTMWcph+1fNTTVZLSJbV/5lLFkFs=";
+
+  meta = with lib; {
+    description = "a replacement for fancontrol with more fine-grained control interface in its config file";
```
Comment (gador) [inline pkgs/os-specific/linux/fanctl/default.nix:17]:
```suggestion
    description = "Replacement for fancontrol with more fine-grained control interface in its config file";
```

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 198350` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>fanctl</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
fanctl:
<details>

got build log for '/nix/store/m46qn13fgz6rsc1vmyc9f4pfr4hr6pib-fanctl-0.6.4' from 'daemon'
When evaluating attribute ‘fanctl’:
warning: unclear-gpl
`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/fanctl/default.nix:19:5:

```
   |
19 |     license = licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: unused-argument
Unused argument: `stdenv`.
Near pkgs/os-specific/linux/fanctl/default.nix:1:3:

```
  |
1 | { stdenv, lib, fetchFromGitLab, rustPlatform, pkgs }:
  |   ^
```
warning: unused-argument
Unused argument: `pkgs`.
Near pkgs/os-specific/linux/fanctl/default.nix:1:47:

```
  |
1 | { stdenv, lib, fetchFromGitLab, rustPlatform, pkgs }:
  |                                               ^
```
</details>
</details>

File: pkgs/os-specific/linux/fanctl/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, lib, fetchFromGitLab, rustPlatform, pkgs }:
```
Comment (gador) [inline pkgs/os-specific/linux/fanctl/default.nix:1]:
```suggestion
{ lib, fetchFromGitLab, rustPlatform }:
```

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 198350` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>fanctl</li>
</ul>
</details>

**Review by gador — APPROVED** [review]

File: pkgs/os-specific/linux/fanctl/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{ lib, fetchFromGitLab, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  version = "0.6.4";
+  pname = "fanctl";
+
+  src = fetchFromGitLab {
+    owner = "mcoffin";
+    repo = pname;
+    rev = "fbb440c520630dff57a8d9923fc0c70ea549caa5";
```
Comment (NickCao) [inline pkgs/os-specific/linux/fanctl/default.nix:10]: Why not use version here?

**Review by NickCao — APPROVED** [review]

**Pull request merged**